File: maximum_subarray/maximum_subarray.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        if (nums.size() == 0) return 0;
        int best = nums[0];
        int cur = 0;

        for (int i = 0; i < nums.size(); i++) {
            if (cur < 0) cur = 0;
            cur += nums[i];
            if (cur > best) best = cur;
        }

        return best;
    }
};

File: maximum_subarray/maximum_subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        cursum = 0
        maxsum = nums[0]

        for i in range(len(nums)):
            if cursum < 0:
                cursum = 0
            cursum += nums[i]
            if cursum > maxsum:
                maxsum = cursum
            
        return maxsum
            